src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/SQLClause.java

```diff
@@ -47,6 +50,12 @@
      */
     abstract void rewrite(Scope scope);
 
+    SQLMethodInvokeExpr replaceByNestedFunction(SQLExpr expr, String nestedPath) {
```

Comment:
np: to me "replace with" sounds more a bit more explicit ("replace by" nested function might be confusing in a sense that the replacement is done by the function).

Reply:
Since I am overloading an existing method of this name, I am keeping it as it for now.